sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```diff
@@ -793,18 +814,28 @@ async def get_throughput(self, **kwargs: Any) -> ThroughputProperties:
         :rtype: ~azure.cosmos.offer.ThroughputProperties
         """
         response_hook = kwargs.pop('response_hook', None)
-        properties = await self._get_properties()
-        link = properties["_self"]
-        query_spec = {
-            "query": "SELECT * FROM root r WHERE r.resource=@link",
-            "parameters": [{"name": "@link", "value": link}],
-        }
-        throughput_properties = [throughput async for throughput in
-                                 self.client_connection.QueryOffers(query_spec, **kwargs)]
-        if len(throughput_properties) == 0:
-            raise CosmosResourceNotFoundError(
-                status_code=StatusCodes.NOT_FOUND,
-                message="Could not find ThroughputProperties for container " + self.container_link)
+        throughput_properties: List[Dict[str, Any]]
+        # In case of a container recreate we will run this a second time
+        for attempt in range(2):
+            if attempt == 1:
+                # Refresh cache if no offer throughput was received the first time
+                await self.read()
+            properties = await self._get_properties()
+            link = properties["_self"]
+            query_spec = {
+                "query": "SELECT * FROM root r WHERE r.resource=@link",
+                "parameters": [{"name": "@link", "value": link}],
+            }
+            throughput_properties = [throughput async for throughput in
+                                     self.client_connection.QueryOffers(query_spec, **kwargs)]
+            if len(throughput_properties) == 0:
+                # Try one more time after refreshing the container properties cache
+                if attempt < 1:
+                    continue
+                raise CosmosResourceNotFoundError(
+                    status_code=StatusCodes.NOT_FOUND,
```

Comment:
If we are going to filter which Substatus of 404 we treat, we might want to add it here, otherwise this is a blank 404/0

Reply:
Adding the comment as a way of tracking. cc: @bambriz

Reply:
A sub status code was added to this. it is now 404/10004